packages/react-sdk/src/index.tsx

```diff
@@ -0,0 +1,68 @@
+import BucketSingleton from "@bucketco/tracking-sdk";
+import type {
+  Flags,
+  Options as BucketSDKOptions,
+  FeatureFlagsOptions,
+} from "@bucketco/tracking-sdk";
+import React, {
+  ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+
+export type BucketInstance = typeof BucketSingleton;
+
+type BucketContext = {
+  bucket: BucketInstance;
+  flags: Flags;
+};
+
+const Context = createContext<BucketContext>({
+  bucket: BucketSingleton,
+  flags: {},
+});
+
+export type BucketProps = BucketSDKOptions & {
+  publishableKey: string;
+  flags: FeatureFlagsOptions;
+  children?: ReactNode;
+  sdk?: BucketInstance;
+};
+
+export default function Bucket({ children, sdk, ...config }: BucketProps) {
+  const [flags, setFlags] = useState<Flags>({});
+  const [bucket] = useState(() => sdk ?? BucketSingleton);
+
+  const contextKey = JSON.stringify(config);
```

Comment:
is this guaranteed to return something stable? We've used json-canonical in the past.

Reply:
Good shout, have used that